tutorial/markdown/java/spring-data/spring-data.md

```diff
@@ -0,0 +1,416 @@
+</dependency>
+<dependency>
+  <groupId>org.springframework.boot</groupId>
+  <artifactId>spring-boot-starter-web</artifactId>
+</dependency>
+<dependency>
+  <groupId>org.springdoc</groupId>
+  <artifactId>springdoc-openapi-ui</artifactId>
+  <version>1.6.6</version>
+</dependency>
+```
+
+#### Useful Links
+
+- [Spring Data Couchbase - Reference Documentation](https://docs.spring.io/spring-data/couchbase/docs/current/reference/html/)
+- [Spring Data Couchbase - JavaDoc](https://docs.spring.io/spring-data/couchbase/docs/current/api/)
+
+### Database Server Configuration
+
+Spring Data couchbase connector can be configured by providing a `@Configuration` [bean](https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-definition) that extends [`AbstractCouchbaseConfiguration`](https://docs.spring.io/spring-data/couchbase/docs/current/api/org/springframework/data/couchbase/config/AbstractCouchbaseConfiguration.html).
+The sample application provides a configuration bean that uses default couchbase login and password:
+
+```java
+@Configuration
+public class CouchbaseConfiguration extends AbstractCouchbaseConfiguration {
+
+  @Override
+  public String getConnectionString() {
+    // capella
+    // return "couchbases://cb.xyz.cloud.couchbase.com";
+
+    // localhost
+    return "127.0.0.1"
+  }
+
+  @Override
+  public String getUserName() {
+    return "Administrator";
+  }
+
+  @Override
+  public String getPassword() {
+    return "password";
+  }
+
+  @Override
+  public String getBucketName() {
+    return "springdata_quickstart";
+  }
+
+  ...
+```
+
+> _from config/CouchbaseConfiguration.java_
+
+This default configuration assumes that you have a locally running Couchbae server and uses standard administrative login and password for demonstration purpose.
+Applications deployed to production or staging environments should use less privileged credentials created using [Role-Based Access Control](https://docs.couchbase.com/go-sdk/current/concept-docs/rbac.html).
+Please refer to [Managing Connections using the Java SDK with Couchbase Server](https://docs.couchbase.com/java-sdk/current/howtos/managing-connections.html) for more information on Capella and local cluster connections.
+
+### Application Properties
+
+You need to configure the connection details to your Couchbase Server in the application.properties file located in the src/main/resources directory.s
+
+```properties
+spring.couchbase.bootstrap-hosts=DB_CONN_STR
+spring.couchbase.bucket.user=DB_USERNAME
+spring.couchbase.bucket.password=DB_PASSWORD
+```
+
+In the connection string, replace `DB_CONN_STR` with the connection string of your Couchbase cluster. Replace `DB_USERNAME` and `DB_PASSWORD` with the username and password of a Couchbase user with access to the bucket.
+
+The connection string should be in the following format:
+
+```properties
+spring.couchbase.bootstrap-hosts=couchbases://cb.fwu-oviwcpjq4v.cloud.couchbase.com
```

Comment:
Please remove the valid connection string from here.

Reply:
changed it